### PR TITLE
[Enhancement] ClusterCaCert - Pass labels to secrets

### PR DESCRIPTION
**Pull request by javiercri**
### Type of change

- Enhancement

### Description

It would be useful to define annotations on cluster ca certificate, which will be passed/copied to the secrets created by the operator.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java:146]: Could you use individual imports instead of this as it was before?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java:382]: I'm not sure this works. First, it might run thousand times because of the parametrized test. So it would be better to add this to some of the existing tests and to the parameters. Also, this actually doesn't test the result. You set it up,  but you never compare the annotations and labels on the secrets.

I will think how to fix the test / or where to run it to make it better.

**Review by scholzj — COMMENTED** [review]
I think the code looks good now. Some comments to the test which I will look on closer. You also have to run the `make crd_install` command to copy the updated CRDs to the right places and fix the build issues .

**Comment by scholzj** [comment]
@javiercri I added a better test for this feature. Please pull the changes from your branch. You will still need to remove your old test and address the other comment.

**Comment by scholzj** [comment]
@javiercri The code looks good to me. But it seems that the checkstyle does not like something and is failing the CIs. Could you have a look?

**Comment by scholzj** [comment]
Ok, the CI got past the checkstyle and into the tests now ... but there seem to be some NPEs:
```
2020-10-20T10:03:49.5925589Z [INFO] 
2020-10-20T10:03:49.5926232Z [INFO] Results:
2020-10-20T10:03:49.5926536Z [INFO] 
2020-10-20T10:03:49.5930906Z [ERROR] Errors: 
2020-10-20T10:03:49.5931601Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext)
2020-10-20T10:03:49.5936749Z [ERROR]   Run 1: CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6024223Z [ERROR]   Run 2: CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6025900Z [ERROR]   Run 3: CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6027038Z [ERROR]   Run 4: CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6028031Z [ERROR]   Run 5: CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6032210Z [ERROR]   Run 6: CertificateRenewalTest.testCustomCertsNotReconciled(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6032735Z [INFO] 
2020-10-20T10:03:49.6033274Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext)
2020-10-20T10:03:49.6034226Z [ERROR]   Run 1: CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6053790Z [ERROR]   Run 2: CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6055266Z [ERROR]   Run 3: CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6064130Z [ERROR]   Run 4: CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6074850Z [ERROR]   Run 5: CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6081225Z [ERROR]   Run 6: CertificateRenewalTest.testExpiredCertsGetRemovedAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6081946Z [INFO] 
2020-10-20T10:03:49.6087882Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext)
2020-10-20T10:03:49.6092961Z [ERROR]   Run 1: CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6098041Z [ERROR]   Run 2: CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6103311Z [ERROR]   Run 3: CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6107956Z [ERROR]   Run 4: CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6114562Z [ERROR]   Run 5: CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6120263Z [ERROR]   Run 6: CertificateRenewalTest.testGenerateTruststoreFromOldSecrets(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6121387Z [INFO] 
2020-10-20T10:03:49.6124794Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext)
2020-10-20T10:03:49.6128231Z [ERROR]   Run 1: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6129645Z [ERROR]   Run 2: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6133619Z [ERROR]   Run 3: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6135001Z [ERROR]   Run 4: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6139031Z [ERROR]   Run 5: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6140385Z [ERROR]   Run 6: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6141275Z [INFO] 
2020-10-20T10:03:49.6145164Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext)
2020-10-20T10:03:49.6149117Z [ERROR]   Run 1: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6152964Z [ERROR]   Run 2: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6154713Z [ERROR]   Run 3: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6160600Z [ERROR]   Run 4: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6164144Z [ERROR]   Run 5: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6169989Z [ERROR]   Run 6: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoOutsideOfMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6170954Z [INFO] 
2020-10-20T10:03:49.6171819Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext)
2020-10-20T10:03:49.6175858Z [ERROR]   Run 1: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6180052Z [ERROR]   Run 2: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6183704Z [ERROR]   Run 3: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6188907Z [ERROR]   Run 4: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6190692Z [ERROR]   Run 5: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6194156Z [ERROR]   Run 6: CertificateRenewalTest.testNewCertsGetGeneratedWhenInRenewalPeriodAutoWithinMaintenanceWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6195341Z [INFO] 
2020-10-20T10:03:49.6201379Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext)
2020-10-20T10:03:49.6292762Z [ERROR]   Run 1: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6293923Z [ERROR]   Run 2: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6294896Z [ERROR]   Run 3: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6307653Z [ERROR]   Run 4: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6309919Z [ERROR]   Run 5: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6325679Z [ERROR]   Run 6: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoOutsideOfTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6326271Z [INFO] 
2020-10-20T10:03:49.6326743Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext)
2020-10-20T10:03:49.6345707Z [ERROR]   Run 1: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6347074Z [ERROR]   Run 2: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6365505Z [ERROR]   Run 3: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6366695Z [ERROR]   Run 4: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6371592Z [ERROR]   Run 5: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6384570Z [ERROR]   Run 6: CertificateRenewalTest.testNewKeyGeneratedWhenInRenewalPeriodAutoWithinTimeWindow(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6385186Z [INFO] 
2020-10-20T10:03:49.6385806Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext)
2020-10-20T10:03:49.6403821Z [ERROR]   Run 1: CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6404876Z [ERROR]   Run 2: CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6423163Z [ERROR]   Run 3: CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6424187Z [ERROR]   Run 4: CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6425304Z [ERROR]   Run 5: CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6443645Z [ERROR]   Run 6: CertificateRenewalTest.testNewKeyGetGeneratedWhenInRenewalPeriodAuto(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6444669Z [INFO] 
2020-10-20T10:03:49.6445126Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext)
2020-10-20T10:03:49.6464786Z [ERROR]   Run 1: CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6466034Z [ERROR]   Run 2: CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6467398Z [ERROR]   Run 3: CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6489355Z [ERROR]   Run 4: CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6490441Z [ERROR]   Run 5: CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6491302Z [ERROR]   Run 6: CertificateRenewalTest.testReconcileCasGeneratesCertsInitially(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6491900Z [INFO] 
2020-10-20T10:03:49.6497491Z [ERROR] io.strimzi.operator.cluster.operator.assembly.CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext)
2020-10-20T10:03:49.6503081Z [ERROR]   Run 1: CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6509037Z [ERROR]   Run 2: CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6514999Z [ERROR]   Run 3: CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6579723Z [ERROR]   Run 4: CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6580940Z [ERROR]   Run 5: CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6584311Z [ERROR]   Run 6: CertificateRenewalTest.testReconcileCasNoCertsGetGeneratedOutsideRenewalPeriod(VertxTestContext) Â» NullPointer
2020-10-20T10:03:49.6584896Z [INFO] 
2020-10-20T10:03:49.6585154Z [INFO] 
2020-10-20T10:03:49.6585473Z [ERROR] Tests run: 6467, Failures: 0, Errors: 11, Skipped: 42
```

**Comment by scholzj** [comment]
Looks like you broke something in the checkstyles again :-/

**Review by scholzj — APPROVED** [review]
I pushed in one more fix. But LGTM now.

**Comment by scholzj** [comment]
@strimzi/maintainers Can someone give it the second review?

**Review by samuel-hawker — APPROVED** [review]
LGTM. Thanks for the contribution.

**Review by tombentley — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**
